### PR TITLE
refactor: win game animation

### PR DESCRIPTION
**Pull request by OropezaHugo**
## Changes Made

- [x] add win game animation
- [x] split the winflag sprites

## Checklist

- [x] I have reviewed my own code changes.
- [x] I have tested these changes locally.
- [x] My code follows the project's coding style guidelines.
- [x] I have updated the documentation accordingly.
- [x] I have added appropriate labels to this pull request.

@CatCoder36 
@JuanPabloArequipa

### CODE REVIEW

**Comment by OropezaHugo** [comment]
move Mario's x position to 12500 if you want to test faster

**Comment by JuanPabloArequipa** [comment]
The win flag is not moving as the original implementation.
Take into account:
- Mario collides with the pole
- Mario go down until collide with the floor
- Flag go down until collide with floor
- Mario shows animation of going to castle

You can check the previous implementation of Gaston

Actual:

https://github.com/Nintentando-Team-Software-Developmet/MarioBrosGame/assets/135129192/3c578457-a565-4a2d-886b-8a0fa3a55ef9

**Comment by JuanPabloArequipa** [comment]
BUG
After arriving to the castle the first time, at the second the animation of entering to the castle is working wrong, Mario stays freeze: (check specifically at second )

https://github.com/Nintentando-Team-Software-Developmet/MarioBrosGame/assets/135129192/a9b33b53-0643-4d38-8372-87fb354bd432

**Review by JuanPabloArequipa — DISMISSED** [review]

File: MarioGame/Source/Systems/PlayerSystem.cs [hunk 1]
```diff
@@ -12,13 +12,21 @@
...
+using Vector2 = nkast.Aether.Physics2D.Common.Vector2;
 
 namespace SuperMarioBros.Source.Systems
 {
     public class PlayerSystem : BaseSystem
     {
         private List<Body> _enemyBodies = new();
         private HashSet<Entity> registeredEntities = new();
+        private List<Body> _poleBodies = new();
+        private bool _teleportMario { get; set; }
```
Comment (CatCoder36) [inline MarioGame/Source/Systems/PlayerSystem.cs:24]: Intenta manejar datos dentro de los componentes, los sistemas deben funcionar para todos las entidades que tengan el componente player, de la manera actual solo estarias controlando el _teleportMario para el primer player que se encuentre, en caso de agregar el player component a otra entidad el sistema dejaria de funcionar, suena inncesario ya que solo hay 1 Mario pero es la logica de ECS

**Review by CatCoder36 — APPROVED** [review]

**Review by JuanPabloArequipa — APPROVED** [review]

**Pull request merged**